### PR TITLE
Update to latest smart contracts

### PR DESCRIPTION
**Pull request by nlordell**
This PR vendors the latest GPv2 Settlement contract artifact. Note that some minor changes were required:
- Rename `allowance_manager()` -> `vault_relayer()`
- Vendor Balancer V2 Authorizer contract (which is needed for Vault deployment, used by the latest Settlement contract version).

### Test Plan

E2E tests are still failing because of changes in rounding in the settlement contract. The issue is addressed in https://github.com/gnosis/gp-v2-services/pull/938.

Verify that the transaction hashes are correct by checking block explorers to see that indeed contract `0x9008D19f58AAbD9eD0D60971565AA8510560ab41` got created:
- [Mainnet](https://etherscan.io/tx/0xf49f90aa5a268c40001d1227b76bb4dd8247f18361fcad9fffd4a7a44f1320d3#internal)
- [Rinkeby](https://rinkeby.etherscan.io/tx/0x609fa2e8f32c73c1f5dc21ff60a26238dacb50d4674d336c90d6950bdda17a21#internal)
- [xDai](https://blockscout.com/xdai/mainnet/tx/0x9ddc538f89cd8433f4a19bc4de0de27e7c68a1d04a14b327185e4bba9af87133/internal-transactions)

### CODE REVIEW

File: contracts/build.rs [hunk 1]
```diff
@@ -93,7 +96,7 @@ fn main() {
             .add_network(
                 "1",
                 Network {
-                    address: addr("0x3328f5f2cEcAF00a2443082B657CedEAf70bfAEf"),
+                    address: addr("0x9008D19f58AAbD9eD0D60971565AA8510560ab41"),
                     deployment_information: Some(tx(
                         "34b7f9a340e663df934fcc662b3ec5fcd7cd0c93d3c46f8ce612e94fff803909",
                     )),
```
Comment (vkgnosis) [inline contracts/build.rs:102]: I think the deployment tx needs to be updated too. Same for the other address changes below.

**Review by vkgnosis — APPROVED** [review]

File: contracts/build.rs (hunk 1 above)
Comment (nlordell) [reply contracts/build.rs:102]: Oops! I forgot.

File: orderbook/src/account_balances.rs [hunk 2]
```diff
@@ -35,7 +35,7 @@ pub trait BalanceFetching: Send + Sync {
 
 pub struct Web3BalanceFetcher {
     web3: Web3,
-    allowance_manager: H160,
+    vault_relayer: H160,
```
Comment (bh2smith) [inline orderbook/src/account_balances.rs:38]: Do we only ever use the vault relayer now? Does this mean we don't need the allowance manager anymore? I wasn't aware of this...

**Review by bh2smith — APPROVED** [review]

File: contracts/artifacts/BalancerV2Authorizer.json [hunk 3]
```diff
@@ -0,0 +1 @@
...r must have ``role``'s admin role."},"grantRoles(bytes32[],address)":{"details":"Grants multiple roles to a single account."},"grantRolesToMany(bytes32[],address[])":{"details":"Grants roles to a list of accounts."},"hasRole(bytes32,address)":{"details":"Returns `true` if `account` has been granted `role`."},"renounceRole(bytes32,address)":{"details":"Revokes `role` from the calling account. Roles are often managed via {grantRole} and {revokeRole}: this function's purpose is to provide a mechanism for accounts to lose their privileges if they are compromised (such as when a trusted device is misplaced). If the calling account had been granted `role`, emits a {RoleRevoked} event. Requirements: - the caller must be `account`."},"revokeRole(bytes32,address)":{"details":"Revokes `role` from `account`. If `account` had already been granted `role`, emits a {RoleRevoked} event. Requirements: - the caller must have ``role``'s admin role."},"revokeRoles(bytes32[],address)":{"details":"Revokes multiple roles from a single account."},"revokeRolesFromMany(bytes32[],address[])":{"details":"Revokes roles from a list of accounts."}},"version":1},"userdoc":{"kind":"user","methods":{},"version":1}}
```
[the remark this answers is not in the document]
Comment (vkgnosis) [reply contracts/artifacts/BalancerV2Authorizer.json:1]: These are generated by the contracts/vendor script. Newlines are usually not *needed* anyway but it is good practice to have them in text files that are edited by humans. For these files it doesn't matter.

File: orderbook/src/account_balances.rs (hunk 2 above)
Comment (nlordell) [reply orderbook/src/account_balances.rs:38]: The allowance manager contract no longer exists in the updated contracts.

Comment (bh2smith) [reply orderbook/src/account_balances.rs:38]: So I guess there was no point in spending so much time making it upgradable...

Comment (nlordell) [reply orderbook/src/account_balances.rs:38]: The allowance manager was never upgradable... I think you may be thinking of authorization contract (i.e. `GPv2AllowListAuthentication`) - which is still there and is still upgradable.

**Comment by nlordell** [comment]
Merging to feature branch as CI was fixed in https://github.com/gnosis/gp-v2-services/pull/938, and keeping the commits separate helps understanding the changes in the future (assuming a **rebase** merge).

**Pull request merged**